tool/internal/ast/shared.go

```diff
@@ -42,15 +43,45 @@ func FindFuncDeclWithoutRecv(root *dst.File, funcName string) *dst.FuncDecl {
 	return decls[0]
 }
 
-func FindFuncDecl(root *dst.File, funcName string) []*dst.FuncDecl {
-	const maxMatchDecls = 2
+func FindFuncDecl(root *dst.File, funcName string, recv string) *dst.FuncDecl {
```

Comment:
hmm, maybe `FindFuncDecl` should just take `*rule.InstFuncRule` as argument

Reply:
I initially thought this was a good idea, but after trying it out, I found that in some call sites, the function name itself is passed directly. This means we're not always looking up the target Func for a given rule; it's also possible that we need to find the corresponding Hook function. So perhaps it's best to keep the current implementation.

https://github.com/open-telemetry/opentelemetry-go-compile-instrumentation/blob/bdef73a5f61240fada5b649b0efb7665964d640a/tool/internal/instrument/trampoline.go#L155-L159